src/GitVersion.App/GitVersionExecutor.cs

```diff
@@ -57,7 +57,7 @@ public int Execute(GitVersionOptions gitVersionOptions)
 
         private int RunGitVersionTool(GitVersionOptions gitVersionOptions)
         {
-            var mutexName = gitVersionOptions.WorkingDirectory.Replace(Path.DirectorySeparatorChar.ToString(), "");
+            var mutexName = repositoryInfo.DotGitDirectory.Replace(Path.DirectorySeparatorChar.ToString(), "");
```

Comment:
Perhaps prefixing `gitversion` to the mutex to avoid anyone else using the same mutex?

Reply:
It seems a bit unlikely that another mutex with the exact name of the path of the .git folder would exist, but it probably wouldn't hurt to be a bit more defensive.

Looking at the mutex documentation, there is some vague stuff about potential maximum lengths of mutex names, but there are no actual values given, so I guess we'll have to see if there are any bug reports about names being too long and determine what do to if that happens.

Reply:
The maximum length of mutex names is `MAX_PATH`, which is defined as 260 characters. So you should be fine here. (Source: [CreateMutexW documentation](https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-createmutexw#parameters))
Please note that mutex names are case sensitive. Should be ok as long as `repositoryInfo.DotGitDirectoryrepositoryInfo.DotGitDirectory` has always the same casing.